command/upgrade/provider.ts

```diff
@@ -84,7 +84,12 @@ export abstract class Provider {
       const { latest } = await this.getVersions(name);
       to = latest;
     }
-    const registry: string = new URL(main, this.getRegistryUrl(name, to)).href;
+
+    const registryUrl = this.getRegistryUrl(name, to)
+    const registry: string =
+      registryUrl.startsWith('jsr:') ?
+      `${registryUrl}/${name}` :
```

Comment:
Name is already added to registryUrl by getRegistryUrl.

```suggestion
      registryUrl :
```

Reply:
or did you mean `${registryUrl}/${main}`?

Reply:
If we use the main option for jsr exports, we need to add a main option to the `Provider` constructor so we can define it differently for different providers. The provider main option would then override the upgrade command main option.